lib/CMakeLists.txt

```diff
@@ -212,7 +224,8 @@ target_compile_options(quda PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,Clang>:--cuda-p
 target_include_directories(quda PRIVATE .)
 target_include_directories(quda PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
 target_include_directories(quda SYSTEM PRIVATE ../include/externals)
-target_include_directories(quda SYSTEM PUBLIC ${CUDAToolkit_INCLUDE_DIRS})
+target_include_directories(quda SYSTEM PUBLIC $<$<OR:$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>,$<COMPILE_LANG_AND_ID:CUDA,Clang>>:${CUDAToolkit_INCLUDE_DIRS}>)
```

Comment:
Is this not needed for NVHPC? Or can we just simplify this to `<COMPILE_ALNG:CUDA>`

Reply:
Oh good catch. This is probably needed, and probably explains some flaky behaviour I've seen (sometime not seeing cufft.h for example).  f242bbcaf